lhotse/recipes/dihard3.py

```diff
@@ -0,0 +1,112 @@
+"""
+About the DIHARD III corpus
+
+    The DIHARD III corpus consists of multi-domain data prepared to evaluate
+    "hard" speaker diarization. It was used for evaluation in the Third DIHARD
+    Challenge, organized by NIST and LDC in Winter 2020. It consists of monologues,
+    map task dialogues, broadcast interviews, sociolinguistic interviews, meeting 
+    speech, speech in restaurants, clinical recordings, and YouTube videos.
+    More details can be found at:
+    https://dihardchallenge.github.io/dihard3/docs/third_dihard_eval_plan_v1.2.pdf
+"""
+import tarfile
+from itertools import chain
+from pathlib import Path
+from typing import Optional, List, Dict, Union
+
+from lhotse import validate_recordings_and_supervisions
+from lhotse.audio import Recording, RecordingSet
+from lhotse.supervision import SupervisionSegment, SupervisionSet
+from lhotse.utils import Pathlike, check_and_rglob, urlretrieve_progress
+
+
+def prepare_dihard3(
+    audio_dir: Pathlike,
+    output_dir: Optional[Pathlike] = None,
+    uem_manifest: Optional[bool] = True,
+) -> Dict[str, Union[RecordingSet, SupervisionSet]]:
+    """
+    Prepare manifests for the DIHARD III corpus.
+    We create two manifests: one with recordings, and the other one with supervisions containing speaker id
+    and timestamps.
+
+    :param audio_dir: Path to downloaded DIHARD III corpus (LDC2020E12), e.g.
+        `/data/corpora/LDC/LDC2020E12_Third_DIHARD_Challenge_Development_Data`
+    :param output_dir: Directory where the manifests should be written. Can be omitted to avoid writing.
+    :param uem_manifest: If True, also return a SupervisionSet describing the UEM segments (see use in
+        dataset.DiarizationDataset)
+    :return: A dict with manifests. The keys are: ``{'recordings', 'supervisions'}``.
+    """
+    audio_paths = check_and_rglob(audio_dir, "*.flac")
+    rttm_paths = list(check_and_rglob(audio_dir, "*.rttm"))
+    uem_paths = list(check_and_rglob(audio_dir, "*.uem"))
+
+    recordings = RecordingSet.from_recordings(
```

Comment:
I recently added `RecordingSet.from_dir` that can do this in parallel if it makes sense